com.mastermindzh.tidal-hifi.yml

```diff
@@ -0,0 +1,62 @@
+---
+app-id: com.mastermindzh.tidal-hifi
+base: org.electronjs.Electron2.BaseApp
+base-version: '21.08'
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: run.sh
+build-options:
+  cflags: '-O2 -g'
+  cxxflags: '-O2 -g'
+  env:
+    NPM_CONFIG_LOGLEVEL: info
+  append-path: /usr/lib/sdk/node16/bin
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.node16
+finish-args:
+  - '--share=ipc'
+  - '--socket=x11'
+  - '--socket=pulseaudio'
+  - '--share=network'
+  - '--device=dri'
+  - '--talk-name=org.freedesktop.Notifications'
+  - '--filesystem=xdg-run/app/com.discordapp.Discord:create'
+  - '--own-name=org.mpris.MediaPlayer2.tidal-hifi'
+  - '--talk-name=org.freedesktop.Notifications'
+modules:
+  - name: tidal-hifi
+    buildsystem: simple
+    sources:
+      - type: git
+        url: https://github.com/Mastermindzh/tidal-hifi
+        tag: 2.8.0
+        commit: 89589b75e1cdbecbeffbbe7933696bc330106d35
+        dest: main
+      - generated-sources.json
+      - type: file
+        path: run.sh
+      - type: file
+        path: com.mastermindzh.tidal-hifi.desktop
+      - type: file
+        path: com.mastermindzh.tidal-hifi.appdata.xml
```

Comment:
these two files should be managed upstream. The lack of .desktop is already a bug as it is the standard for automatically have a desktop app available into the DE.

Neither are specific to flatpak either.

Reply:
The desktop file is different from the upstream one because the flatpak version requires `run.sh` whilst the upstream does not.

I don't understand why appdata.xml should be managed upstream, it belongs to the release as a flatpak.

Reply:
> The desktop file is different from the upstream one because the flatpak version requires `run.sh` whilst the upstream does not.

You can use the `desktop-file-edit` command to change that in the `post-install`

> I don't understand why appdata.xml should be managed upstream, it belongs to the release as a flatpak.

No it is NOT specfic to flatpak.

Reply:
If you can provide an example or docs for `desktop-file-edit` I will be able to take a look.

As for the appdata/metainfo file, please see [The appdata guidelines](https://github.com/flathub/flathub/wiki/AppData-Guidelines).
Alternatively, if you have information that gives additional insights I'm willing to take a look there as well.

PS: I renamed appdata to metainfo as per the guidelines.

Reply:
> If you can provide an example or docs for `desktop-file-edit` I will be able to take a look.

I don't know more than `man desktop-file-edit` can tell you.